src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2022 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jvnet.hudson.test;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.model.Slave;
+import hudson.remoting.Launcher;
+import hudson.remoting.Which;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.JNLPLauncher;
+import hudson.slaves.RetentionStrategy;
+import hudson.util.StreamCopyThread;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.tools.ant.util.JavaEnvUtils;
+import org.junit.rules.ExternalResource;
+
+/**
+ * Manages inbound agents.
+ * While these run on the local host, they are launched outside of Jenkins.
+ * @see JenkinsRule#createComputerLauncher
+ * @see JenkinsRule#createSlave()
+ */
+public final class InboundAgentRule extends ExternalResource {
+
+    private final Map<String, Process> procs = new HashMap<>();
+
+    /**
+     * Creates, attaches, and starts a new inbound agent.
+     * @param name an optional {@link Slave#getNodeName}
+     */
+    public Slave createAgent(@NonNull JenkinsRule r, @CheckForNull String name) throws Exception {
+        if (name == null) {
+            name = "agent" + r.jenkins.getNodes().size();
+        }
+        // TODO 2.216+ support WebSocket option
+        DumbSlave s = new DumbSlave(name, new File(r.jenkins.getRootDir(), "agent-work-dirs/" + name).getAbsolutePath(), new JNLPLauncher(true));
+        s.setNumExecutors(1); // TODO pending 2.234+
```

Comment:
seems reasonable to bump it?

Reply:
Possibly. Seems the `plugin-pom` currently expects 2.249+ but it is unclear to me whether this is a hard requirement:

> `jenkins.version>2.249.1` is expected to be used for most plugins.

Reply:
We don't have a formal policy, but I think a de facto policy of waiting until the given core disappears from the Update Center (currently 500 days) is a good enough policy as any.

Reply:
Seems reasonable. I do not see any particular reason to complicate this PR with such a bump, since it works fine as it is and the edits to clean up with a new core are very simple.

Reply:
I suppose they might be simple, but they were not obvious to me. In the past you have written comments like "TODO 2.xxx+ [newer API]" but I notice that in this PR the future work is left as an exercise for the reader.